napari/_qt/dialogs/qt_layer_info_dialog.py

```diff
@@ -0,0 +1,42 @@
+from qtpy.QtWidgets import QDialog, QHBoxLayout, QLabel, QVBoxLayout
+
+from ...utils.translations import trans
+
+
+class QtLayerInfoDialog(QDialog):
+    """Layer information dialog that can be accessed when right clicking
+    layer on layer list."""
+
+    # resized = Signal(QSize)
+
+    def __init__(self, layer, parent=None):
+
+        super().__init__(parent)
+
+        self.setWindowTitle(trans._("Layer Information"))
+        name = QLabel(layer.name)
+        path = QLabel(layer.source.path)
+        plugin = QLabel(layer.source.reader_plugin)
+        sample = QLabel(str(layer.source.sample))
+        widget = QLabel(str(layer.source.widget))
```

Comment:
I believe these three options are mutually exclusive, and you would never have both a `plugin` and a `sample` entry, for example. So maybe rather than showing the `plugin` in the tooltip we show the source that isn't `None` and then in brackets where the source came from e.g. for your example the tooltip would be `name: test2, source: builtins (plugin)` or something like that.

We would have to make a decision about what we show for the sample source, which shows both the plugin name and the sample item name (I'd say show both), and for the widget, which is actually a function GUI object - there I would say we use `layer.source.widget._function.__name__`